### PR TITLE
feat: Use ENS in profile

### PR DESCRIPTION
**Pull request by Agupane**
# Description

- Replaces ens name and avatar when the ens is registered for that user
- Several improvements on the badge profile

### CODE REVIEW

File: src/components/helpers/Address.tsx [hunk 1]
```diff
@@ -96,7 +98,11 @@ export const Address: React.FC<Props> = ({
   return (
     <Wrapper {...restProps}>
       <ExternalLink href={isUserAddress ? getProfileUrl() : getExplorerUrl(address)}>
-        {truncate ? truncateStringInTheMiddle(address, 8, 6) : address}
+        {isEnsName
```
Comment (FedeMadoery) [inline src/components/helpers/Address.tsx:101]: Can you move this into a more simple cost declaration outside of the return?

```
const displayAddress = truncate  ? truncateStringInTheMiddle(address, 8, 6) : address;
const userIndentified = isEnsName ? ensNameOrAddress : displayAddress

```

File: src/pagePartials/badge/preview/BadgeOwnedPreview.tsx [hunk 2]
```diff
@@ -37,24 +48,67 @@ export default function BadgeOwnedPreview() {
...
   const badgeModel = badge?.badgeModel
+  const creatorAddress = isThirdParty ? badgeModel?.creator.id : null
   const badgeModelMetadata = badgeModel?.badgeModelMetadata
+  const resCreatorMetadata = useS3Metadata<{ content: CreatorMetadata }>(badgeModel?.uri || '')
+  const creatorMetadata = resCreatorMetadata.data?.content
+  let issuer = 'TheBadge'
+  if (isThirdParty && creatorMetadata && creatorMetadata.name) {
+    issuer = creatorMetadata.name
+  }
+  console.log('asd', issuer)
```
Comment (FedeMadoery) [inline src/pagePartials/badge/preview/BadgeOwnedPreview.tsx:66]: 🧹

File: src/pagePartials/badge/preview/BadgeOwnedPreview.tsx [hunk 3]
```diff
@@ -37,24 +48,67 @@ export default function BadgeOwnedPreview() {
...
+  const { web3Provider } = useWeb3Connection()
   const badgeById = useBadgeById(badgeId)
+  const isThirdParty = useIsThirdPartyBadge(badgeId)
+  const theBadge = useContractInstance(TheBadge__factory, 'TheBadge')
 
   const badge = badgeById.data
   const badgeModel = badge?.badgeModel
+  const creatorAddress = isThirdParty ? badgeModel?.creator.id : null
   const badgeModelMetadata = badgeModel?.badgeModelMetadata
+  const resCreatorMetadata = useS3Metadata<{ content: CreatorMetadata }>(badgeModel?.uri || '')
```
Comment (FedeMadoery) [inline src/pagePartials/badge/preview/BadgeOwnedPreview.tsx:60]: are you sure that you have to use `badgeModel?.uri` and not the `creatorAddress` and do a query to the graph?

Comment (Agupane) [reply src/pagePartials/badge/preview/BadgeOwnedPreview.tsx]: yes you were right, its fixed now

**Review by FedeMadoery — APPROVED** [review]
Awesome!! Add ENS was really needed 🚀

**Pull request merged**